src/prover_disk.hpp

```diff
@@ -238,10 +253,78 @@ class DiskProver {
     std::string filename;
     std::vector<uint8_t> memo;
     std::vector<uint8_t> id;  // Unique plot id
-    uint8_t k;
+    uint8_t k{0};
     std::vector<uint64_t> table_begin_pointers;
     std::vector<uint64_t> C2;
 
+    void Read(std::vector<uint8_t>& id_in, std::vector<uint8_t>& memo_in, uint8_t& k_in, std::vector<uint64_t>& table_begin_pointers_in, std::vector<uint64_t>& C2_in) const
```

Comment:
the names of these parameters confuse me. The `_in` suffix suggests they are *in*parameters, but they all appear to be output parameters. Did you consider returning them in a tuple, as multiple return values?

Reply:
I feel like a tuple would make this more confusing than better. See a9ca08363e61ac45ed9c3ccae7840464818a7344.